### ISSUE TITLE
modify asa dataset to use indexing instead of one hot encoding

### ISSUE THREAD

**Issue by RuiFilipeCampos** [Digital-Defiance/llm-voice-chat#29]
cross entropy loss function accepts class indexes so this saves a lot of memory
